x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformTaskTests.java

```diff
@@ -523,6 +528,85 @@ public void testDeriveBasicCheckpointingInfoWithIndexer() {
         assertThat(checkpointingInfo.getNext().getCheckpointProgress(), sameInstance(progress));
     }
 
+    public void testInitializeIndexerWhenAlreadyInitialized() {
+        var transformTask = createTransformTask(
+            TransformConfigTests.randomTransformConfigWithoutHeaders(),
+            MockTransformAuditor.createMockAuditor()
+        );
+        transformTask.initializeIndexer(mock(ClientTransformIndexerBuilder.class));
+        IllegalStateException e = expectThrows(
+            IllegalStateException.class,
+            () -> transformTask.initializeIndexer(mock(ClientTransformIndexerBuilder.class))
+        );
+        assertThat(e.getMessage(), containsString("The object cannot be set twice!"));
+    }
+
+    public void testTriggeredIsNoOpWhenTransformIdMismatch() {
+        var transformId = randomAlphaOfLengthBetween(1, 10);
+        var transformTask = createTransformTask(
+            TransformConfigTests.randomTransformConfigWithoutHeaders(transformId),
+            MockTransformAuditor.createMockAuditor()
+        );
+        transformTask.triggered(new TransformScheduler.Event("not-" + transformId, randomNonNegativeLong(), randomNonNegativeLong()));
```

Comment:
If we wanted to verify this returns without calling the indexer, we could do something like:
```java
var indexer = mock(ClientTransformIndexerBuilder.class);
transformTask.initializeIndexer(indexer);
transformTask.triggered(new TransformScheduler.Event("not-" + transformId, randomNonNegativeLong(), randomNonNegativeLong()));
verifyNoInteractions(indexer);
```

Reply:
Great idea!

Done.